forestbot.lua

```diff
@@ -5,318 +5,141 @@
+                  "handlerModules.movement",
+--                  "handlerModules.effects",
+--                  "handlerModules.stance",
+                }
+
+-- Forward declarations to allow these functions to be private and defined
+-- after functions that call them.
+local reloadModule
+local initModule
+local getModuleFromName
+local reset
 
 --------------------------------------------------------------------------------
 -- Initializer function.  Will be executed when this script file is loaded.
 --------------------------------------------------------------------------------
 function bot.init()
-  if bot.debug then
-    bot.debugMessage = function(s)
-      print(">> " .. s)
-    end
-  else
-    bot.debugMessage = function(s) end
-  end
-
-  bot.debugMessage("bot.init()")
+  debugMessage("bot.init()")
 
-  if not savedPackagePath then
-    savedPackagePath = package.path
-    bot.debugMessage("Capturing default package path")
-    package.path = os.getenv("forestbot_path") .. "/?.lua;" .. savedPackagePath
+  for i, moduleName in ipairs(modules) do
+    reloadModule(bot, moduleName)
+    initModule(bot, moduleName, bot.status)
   end
-
-  -- unload+reload modules here
-  package.loaded["botbtree"] = nil
-  bot.botbtree = require("botbtree")
-  package.loaded["behaviourtree.behaviourtree"] = nil
-  behaviourtree = require("behaviourtree.behaviourtree")
-
-  bot.initHandlers()
-  bot.reset()
-end
-
---------------------------------------------------------------------------------
--- Evaluate the behaviour tree based on the current known state of the world.
---------------------------------------------------------------------------------
-function bot.think()
-  bot.debugMessage("Thinking...")
 end
 
 --------------------------------------------------------------------------------
--- Resets all bot state to initial values.
+-- Calls each module's reset() function, resetting the bot to its initial state.
 --------------------------------------------------------------------------------
 function bot.reset()
-  bot.debugMessage("bot.reset()")
-
-  bot.status = {}
-  bot.status.hits = 0
-  bot.status.energy = 0
-  bot.status.moves = 0
-
-  bot.status.maxHits = 0
-  bot.status.maxMoves = 0
-
-  bot.status.level = 0
-  bot.status.xp = 0
-
-  bot.status.stance = ""
-  
-  bot.location = {}
-  bot.location.roomNo = 0
-
-  bot.needs = {}
-  bot.needs.hunger = 0
-  bot.needs.thirst = 0
-
-  bot.items = {}
-  bot.items.coins = 0
-  bot.items.weight = 0
-  bot.items.wornWeight = 0
-  bot.items.encumbrance = ""
-  bot.items.hasFood = true 
-  bot.items.hasWater = true
-
-  bot.items.inventory = {}
-  bot.items.equipment = {}
-
-  bot.combat.targets = {}
-
-  -- should probably init inventory here
-  -- and stats
-
-  -- reset behaviours
-  bot.btree = bot.botbtree.loadJSON("behaviour.json")
-
-  bot.btree:run(bot)
+  debugMessage("bot.reset()")
 
+  for i, moduleName in ipairs(modules) do
+    resetModule(moduleName)
```

Comment:
The function is at: [Line 104](https://github.com/ProgrammerHero/forestbot/blob/7a6e25c634a817be5e0edfe5114176814322d26e/forestbot.lua#L104). The idea behind these functions is that reload() and init() are used for a full reload and initialization of the modules to allow things like newly-written functions to be loaded. reset() is intended to set the bot's state variables to sensible defaults without changing any of the functions/modules etc.